### PR TITLE
improve-test-coverage-for-admin-portal-directives

### LINKED ISSUE #2149: Improve Test Coverage for Admin Portal Directives
The frontend code coverage report shows 65.07% statement coverage for src/app/directives in the Admin Portal, below the target range of 70–80%.
Goal: Increase statement coverage to 80%.
Scope:
Write unit tests for directive logic, focusing on untested statements and edge cases.

### PR DESCRIPTION
**Pull request by EhsanEhrari**

### CODE REVIEW

File: ui/admin-portal/src/app/directives/highlight-search.directive.spec.ts [hunk 1]
```diff
@@ -28,7 +27,7 @@ import {SearchQueryService} from "../services/search-query.service";
 @Component({
   template: `
     <div appLowercase></div>
-    <div class="highlight" appHighlightSearch>Angular</div>
+    <div class="highlight" appHighlightSearch>Angular is amazing. Learn Angular today!</div>
```
Comment (sadatmalik) [inline ui/admin-portal/src/app/directives/highlight-search.directive.spec.ts:30]: :-)

**Review by sadatmalik — APPROVED** [review]

**Review by hiba-machfej — APPROVED** [review]
